src/systems/filecoin_files/data_transfer/data_transfer_subsystem.id

```diff
@@ -1,66 +1,127 @@
-type Address struct {}
-type Storage struct {}
-type StorageDealID struct {}
-type Function struct {}
-type Piece struct {}
+import ipld "github.com/filecoin-project/specs/libraries/ipld"
+import libp2p "github.com/filecoin-project/specs/libraries/libp2p"
 
-type DataTransferSubsystem struct @(mutable) {
-    dataTransfers  {UInt: DataTransferChannel}
-    volumes        [Storage]
+import piece "github.com/filecoin-project/specs/systems/filecoin_files/piece"
+
+type StorageDeal struct {}
+type RetrievalDeal struct {}
+
+// A DataTransferVoucher is used to validate
+// a data transfer request against the underlying storage or retrieval deal
+// that precipitated it
+type DataTransferVoucher union {
+	StorageDealVoucher 
+	RetrievalDealVoucher
+}
+
+type StorageDealVoucher struct {
+	deal StorageDeal
+}
+
+type RetrievalDealVoucher struct {
+	deal RetrievalDeal 
+}
 
-    OpenDataTransferChannel(x StorageDealID) UInt
-    CloseDataTransferChannel(x UInt)
-    TransferChannelStatus(x UInt) DataTransferStatus
-    ReportDataTransferEvent(x Function) union {Ret DataTransferStatus, Err error}
+type Ongoing struct {}
+type Completed struct {}
+type Failed struct {}
+type ChannelNotFoundError struct {}
 
-    findData(x CID) Piece
+type DataTransferStatus union {
+	Ongoing
+	Completed
+	Failed
+	ChannelNotFoundError
 }
 
-// Assumes access to error-checked, ordered, reliable transmission protocol
-type Scheduler struct {
-    ScheduleTransfer(x UInt, y Piece)
-    requestData(x DataTransferChannel) union {Ret Bytes, Err error}
-    sendData(x DataTransferChannel, y Bytes) error?
+type TransferID UInt
+
+type ChannelID struct {
+	to libp2p.PeerID
+	id TransferID
 }
 
+// All immutable data for a channel
 type DataTransferChannel struct {
-    channelID      UInt
-    contentID      CID
-    dataIncrement  UVarint
-    sender         Address
-    recipient      Address
-    totalSize      UVarint
-    sent           UVarint
-    received       UVarint
-
-    channelType()  DataTransferType  @(cached)
-    transferNum()  Float             @(cached)
+	// an identifier for this channel shared by request and responder, set by requestor through protocol
+	transferID     TransferID
+	// base CID for the piece being transferred
+	PieceRef      CID
```

Comment:
can we call this just `RootCID` ? or `RequestCID` or just `CID`? (trying to remove "piece" concept from DT to make it more fundamental)